core-library/src/main/java/org/silverpeas/core/util/PdfUtil.java

```diff
@@ -45,6 +50,71 @@
  * Date: 08/07/13
  */
 public class PdfUtil {
+  
+  private PdfUtil() {
+    throw new IllegalStateException("Utility class");
+  }
+
+  /**
+   * Gets some document info from a PDF file.
+   * @param pdfSource the source pdf file, this content is not modified by this method
+   * @return a {@link DocumentInfo} instance.
+   */
+  public static DocumentInfo getDocumentInfo(File pdfSource) {
+    if (pdfSource == null || !pdfSource.isFile()) {
+      throw new SilverpeasRuntimeException("The pdf source file doesn't exist");
+    } else if (!FileUtil.isPdf(pdfSource.getPath())) {
+      throw new SilverpeasRuntimeException("The source is not a pdf file");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "The source is not a pdf file" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)